app/Http/Controllers/ServicesUiController.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Service;
+use App\Models\User;
+use App\Repositories\ServicesRepository;
+use Illuminate\Http\Request;
+
+/**
+ * Controller for the UI with extended models
+ */
+class ServicesUiController extends Controller
+{
+    /**
+     * @param ServicesRepository $services
+     */
+    public function __construct(ServicesRepository $services)
+    {
+        $this->services = $services;
+        $this->middleware('auth:api');
+    }
+
+    /**
+     * Get all entities
+     *
+     * Display an extended listing of the resource for the ui.
+     *
+     * @return App\Models\Service
+     */
+    public function index(Request $request)
+    {
+
+        if ($request->user('api')->hasRole('Admin')) {
+            return app('ServicesRepository')->getExpandedServices();
```

Comment:
Why app('ServicesRepository') and not $this->services?

Reply:
no reason whatsoever